pkg/k8s/endpoints.go

```diff
@@ -164,7 +165,16 @@ func ParseEndpointSliceV1Beta1(ep *slim_discovery_v1beta1.EndpointSlice) (Endpoi
 		// unknown state as ready.
 		// More info: vendor/k8s.io/api/discovery/v1beta1/types.go
 		if sub.Conditions.Ready != nil && !*sub.Conditions.Ready {
-			continue
+			if option.Config.EnableK8sTerminatingEndpoint {
+				// Terminating indicates that the endpoint is getting terminated.
+				// Ready is never true when an endpoint is terminating.
+				// Propagate the terminating endpoint state so that we can gracefully
+				// remove those endpoints.
+				// More details : vendor/k8s.io/api/discovery/v1beta1/types.go
+				if sub.Conditions.Terminating != nil && !*sub.Conditions.Terminating {
```

Comment:
Aren't we adding non-ready endpoint if it is in non-terminating state?

Reply:
No, we are not. There is a `continue` on the next line.

Reply:
- We won't invoke `continue` if `--enable-k8s-terminating-endpoints=false` for non-ready endpoints. Is it OK?
- We won't invoke `continue` for non-ready terminating endpoints. Same question.

Reply:
>We won't invoke continue if --enable-k8s-terminating-endpoints=false for non-ready endpoints. Is it OK?

Ah, yes! Need to fix the case where user explicitly disables the feature flag. 

>We won't invoke continue for non-ready terminating endpoints. Same question.

Yes, as the comment indicates, we want to propagate the terminating state.